src/components/common/Input.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { ChangeEvent, InputHTMLAttributes } from 'react';
+import styled from '@emotion/styled';
+import { css } from '@emotion/react';
+
+type Sort = 'left' | 'middle';
+
+interface InputProps extends Omit<InputHTMLAttributes<HTMLInputElement>, 'maxLength'> {
+  className?: string;
+  sort: Sort;
+  valid?: boolean;
+  maxLength?: number;
+  [key: string]: any;
+}
+function Input({ className, sort, valid = true, maxLength, onChange, ...props }: InputProps) {
+  const handleChangeWithMaxLength = (e: ChangeEvent<HTMLInputElement>) => {
+    if (String(e.target.value).length <= maxLength && typeof e.target.value === 'string') {
+      onChange(e);
+    }
+  };
+
+  return (
+    <>
+      <Styled.Input
+        className={className}
+        sort={sort}
+        valid={valid}
+        onChange={maxLength ? handleChangeWithMaxLength : onChange}
+        {...props}
+      />
+    </>
+  );
+}
+
+export default Input;
+
+const Styled = {
+  Input: styled.input<InputProps>`
+    display: block;
+    background: transparent;
+    outline: 0;
+    border: 0;
+    width: 100%;
+    border-bottom: 1px solid #565656;
+    padding-bottom: 10px;
+    color: #565656;
+    &:focus {
+      color: white;
+      border-bottom: ${({ valid }) => (valid ? '1px solid #FFFFFF' : '1px solid #ffc5c5')};
+    }
+    ${({ sort }) =>
+      sort === 'middle'
+        ? css`
+            text-align: center;
+          `
+        : css`
+            text-align: left;
+          `}
+    &::placeholder {
```

Comment:
밑 방식 vs css 방식

혹시 css 방식으로 쓰신 이유가 있을까요?

```suggestion
    text-align: ${({sort}) => sort === 'middle' ? 'center' : 'left'}
    &::placeholder {
```

Reply:
오,,, 그냥 막 갈겼습니다. 밑 방식이 좀 더 깔끔할 것 같군요 ! 수정하겠습니다 : )